specification/offazurespringboot/resource-manager/Microsoft.OffAzureSpringBoot/preview/2024-04-01-preview/springbootdiscovery.json

```diff
@@ -0,0 +1,2072 @@
+                "description": "The machine ARM resource Id of this app instance"
+              },
+              "instanceCount": {
+                "type": "integer",
+                "format": "int32",
+                "description": "The instance count of this app instance"
+              },
+              "jvmMemoryInMB": {
+                "type": "integer",
+                "format": "int32",
+                "description": "The jvm heap memory allocated of this app instance"
+              }
+            },
+            "required": [
+              "machineArmId"
+            ]
+          },
+          "description": "The breakdown info for app instances on all the servers"
+        },
+        "runtimeJdkVersion": {
+          "type": "string",
+          "description": "The jdk version installed on server"
+        },
+        "servers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The server list the app installed"
+        },
+        "machineArmIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The machine ARM id list the app belongs to."
+        },
+        "springBootVersion": {
+          "type": "string",
+          "description": "The spring boot version."
+        },
+        "staticContentLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The static content location list."
+        },
+        "connectionStrings": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The connection string list."
+        },
+        "lastModifiedTime": {
+          "description": "Time when this springbootapps jar file was last modified.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "lastUpdatedTime": {
+          "description": "Time when this springbootapps instance was last refreshed.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource provisioning state.",
+          "enum": [
+            "Unknown",
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Accepted",
+            "Provisioning",
+            "Deleting"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "errors": {
+          "description": "The list of errors.",
+          "type": "array",
+          "items": {
+            "description": "The error",
+            "$ref": "#/definitions/Error"
+          }
+        },
+        "labels": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
```

Comment:
use of AdditionalProperties like these are not allowed : [https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-additional-properties.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/avoid-additional-properties.md)

Why do you need these ? Can you model these with a stricter schema ?

Reply:
Azure Migrate Labels related closures:
=================================

1. ARM team will deliver TagsV2 towards the end of 2024. TagsV2 will support Top-level tags on proxy resources. 
2. Until the Tags support officially rolls out for proxy resources, Azure Migrate team and its partners will add a field named "labels" inside the RP owned properties.
3. ARM Tags PM team and Azure Migrate PM team have agreed upon this contract. [Working Crew: @Shikher Saluja, @Rich Thorn, @Akif Heren, @Jin Soon Lim, @Venkatesh Gajula]
4. Partners of Azure Migrate that will add "labels" to their proxy resources: Oracle, Springboot, MySql
5. Requesting an exception to let the "labels" field's schema be Dictionary<string, string>, as this field indeed captures the tags input by the user.